docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+                        ...
+                },
+                "options": {
+                    ...
+                },
+                "rules": {
+                    "requireIf": string, /* LORIS Logic Parser formula */
+                    "displayIf": string, /* LORIS Logic Parser formula */
+                    "disableIf": string, /* LORIS Logic Parser formula */
+                }
+        ...
+}
+```
+
+A field object has a unique key that is its identifier (ID) and name. The ID will also be used as an identifier for a
+database column or a front-end element. Within the field object, there are 4 required keys: type, description, options,
+and rules.
+
+`type`: Required string. It represents the field's data type i.e. integer, string, boolean, etc.
+
+This will be the data type of the field's value within the `data` component upon collection:
+
+```js
+{
+    "data" : {
+        "nameOfField" : `schema.fields.nameOfField.type`
+    }
+}
+```
+
+`description`: Required string. The human readable description of this field, provided in every available language. It
+may be rendered on the front-end as a question, or label, text.
+
+`options`: Required object of type-specific keys. It contains the type-dependent options for this field, further
+described in section [3.2: Field Types](#32-field-types).
+
+`rules`: Required object with keys whose value may be an empty string.
+
+## 3.1: Rules
+
+The `rules` object contains 3 rules for validation and display: requireIf, displayIf, and disableIf.
+
+```js
+{
+    "schema" : {
+        ...
+        "fields" : {
+            "nameOfField" : {
+                ...
+                "rules": {
+                    "requireIf": string, /* LORIS Logic Parser formula */
+                    "displayIf": string, /* LORIS Logic Parser formula */
+                    "disableIf": string, /* LORIS Logic Parser formula */
+                }
+        ...
+}
+```
+
+The value of these keys are string expressions of logical formulas. These formulas, when calculated, evaluate to true
+or false on the front-end. The formulas are parsed and calculated using the Evaluator function of the LORIS Logic
+Parser, [LParse](https://gitlab.com/zainvirani/LParse). The currently supported logic operations are available in the LParse [README.md](https://gitlab.com/zainvirani/LParse/blob/master/README.md), as well as
+instructions on how to customize functions. 
+
+`rules.requireIf`: String. True or false that the field requires data input, given the condition in the string expression.
+
+`rules.displayIf`: String. True or false that the field is displayed in the instrument, given the condition in the string
+expression.
+
+`rules.disableIf`: String. True or false that the field is disabled i.e. not editable nor submittable, given the condition in
+the string expression.
+
+## 3.2: Field types
+
+There are 8 different field types: enum, string, int, decimal, date, time, boolean, and score. Each type
```

Comment:
I'd say numeric not number but agree

@driusan any ideas to replace enum ? list ? set ?

Reply:
I don't have any problems with enum being a type.. "list" seems like a UI element that displays enumerations, and "sets" are unordered.

Reply:
int, decimal and date, time are grouped further below in the spec into numeric and 'Date and Time'. but they are different data types. if we have a field of type numeric, how else can we validate the data if we don't specify that the data is of type int or decimal? i think it makes sense to group them together, but not coalesce them into the same field type. 

from my research, the datetime input element is about to be unsupported. so if we do want to input a datetime, it would have to be two different fields of date and time. thoughts?